src/modules/inspect/token-viewer/token-viewer.ts

```diff
@@ -2,40 +2,48 @@ import {bindable, inject} from 'aurelia-framework';
 
 import {IShape} from '@process-engine/bpmn-elements_contracts';
 import {DataModels} from '@process-engine/management_api_contracts';
-import {IDiagram} from '@process-engine/solutionexplorer.contracts';
 
+import {IDiagram} from '@process-engine/solutionexplorer.contracts';
 import {ISolutionEntry} from '../../../contracts';
-import {IInspectCorrelationService} from '../inspect-correlation/contracts/index';
 import {
   IPayloadEntry,
   IPayloadEntryValue,
   IRawTokenEntry,
   ITokenEntry,
+  ITokenViewerService,
 } from './contracts/index';
 
-@inject('InspectCorrelationService')
+// tslint:disable: no-magic-numbers
+
+const versionRegex: RegExp = /(\d*)\.(\d*).(\d*)/;
+
+@inject('TokenViewerService')
 export class TokenViewer {
-  @bindable() public correlation: DataModels.Correlations.Correlation;
+
+  @bindable({changeHandler: 'processInstanceIdOrCorrelationChanged' }) public correlation: DataModels.Correlations.Correlation;
   @bindable() public activeDiagram: IDiagram;
   @bindable() public activeSolutionEntry: ISolutionEntry;
   @bindable() public flowNode: IShape;
   @bindable() public token: string;
   @bindable() public showBeautifiedToken: boolean = true;
+  @bindable({changeHandler: 'processInstanceIdOrCorrelationChanged' }) public processInstanceId: string;
+
   public tokenEntries: Array<ITokenEntry> = [];
   public showTokenEntries: boolean = false;
   public firstElementSelected: boolean = false;
   public shouldShowFlowNodeId: boolean = false;
   public rawTokenEntries: Array<IRawTokenEntry>;
 
-  private _inspectCorrelationService: IInspectCorrelationService;
+  private _tokenViewerService: ITokenViewerService;
+  private _getTokenHistoryGroup: Promise<DataModels.TokenHistory.TokenHistoryGroup>;
 
-  constructor(inspectCorrelationService: IInspectCorrelationService) {
-    this._inspectCorrelationService = inspectCorrelationService;
+  constructor(tokenViewerService: ITokenViewerService) {
+    this._tokenViewerService = tokenViewerService;
   }
 
-  public correlationChanged(): void {
-    const correlationWasInitiallyOpened: boolean = this.flowNode === undefined;
-    if (correlationWasInitiallyOpened) {
+  public processInstanceIdOrCorrelationChanged(): void {
+    const noFlowNode: boolean = this.flowNode === undefined;
```

Comment:
You can improve this variable name

```suggestion
    const noFlowNodeSelected: boolean = this.flowNode === undefined;
```